modules/transport.js

```diff
@@ -376,40 +376,42 @@ Transport.prototype.onUnconfirmedTransaction = function(
  * @returns {setImmediateCallback} cb
  */
 Transport.prototype.broadcastHeaders = cb => {
-	// Grab list of all connected peers
-	modules.peers.list({ normalized: false }, (err, peers) => {
-		if (!peers || peers.length === 0) {
-			library.logger.debug('Transport->broadcastHeaders: No peers found');
-			return setImmediate(cb);
-		}
+	// Grab a random list of connected peers.
+	const peers = library.logic.peers.listRandomConnected({
+		limit: constants.maxPeers,
+	});
```

Comment:
In the previous change where you moved `modules.peers.list` to `logic.peers.listRandomConnected`, you also called `findGoodPeers`, which provide you peers with maximum height in the network. 

Here we are making this change without `findGoodPeers`, so bad peers which are not up to network level will also be used. Is that really necessary? 

`broadcastHeaders` is used during `sync` and then `processBlock`, We can remove it from `sync` as it will eventually use `processBlock` in the end.

Reply:
@nazarhussain We shouldn't remove `broadcastHeaders` from sync, as `processBlock` from sync is called with `broadcast = false` so it will not call `broadcastHeaders`.